pkg/controller/instance/instance_controller.go

```diff
@@ -258,6 +276,33 @@ func (r *Reconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 	return computeTheReconcileResult(instance, time.Now), nil
 }
 
+func ensureReadinessInitialized(i *kudoapi.Instance) {
+	if i.Spec.PlanExecution.PlanName == kudoapi.DeployPlanName || i.Spec.PlanExecution.PlanName == kudoapi.UpgradePlanName || i.Spec.PlanExecution.PlanName == kudoapi.UpdatePlanName {
+		i.SetReadinessUnknown()
+	}
+}
+
+func readinessChanged(instance *kudoapi.Instance, instance2 *kudoapi.Instance) bool {
+	ready := meta.FindStatusCondition(instance.Status.Conditions, "Ready")
+	ready2 := meta.FindStatusCondition(instance2.Status.Conditions, "Ready")
+
+	return ready != ready2 && (ready == nil || ready2 == nil || !reflect.DeepEqual(*ready, *ready2))
```

Comment:
Why can't we just
```
return !reflect.DeepEqual(*ready, *ready2)
```
?

Reply:
Good point, `DeepEqual` checks for `nil` values as well.